x-pack/plugins/fleet/server/services/fleet_server/index.ts

```diff
@@ -52,17 +57,54 @@ export async function allFleetServerVersionsAreAtLeast(
     }
   }
 
-  const versionCounts = await getAgentVersionsForAgentPolicyIds(esClient, [...policyIds]);
-  const versions = Object.keys(versionCounts);
+  const managedAgentPolicies = await agentPolicyService.getAllManagedAgentPolicies(soClient);
+  const fleetServerAgents = await getAgentsByKuery(esClient, soClient, {
+    showInactive: true,
```

Comment:
I think this condition has unintended consequences for on-prem fleet-servers, as we only filter out offline/unenrolled fleet-servers for managed policies.
I tested this with an unenrolled fleet-server on version 8.9.0, the secret storage is never enabled.

Reply:
Logic has been updated to account for this case and FTR tests have been added for it as well.